crates/ra_assists/src/add_new.rs

```diff
@@ -0,0 +1,353 @@
+use crate::{Assist, AssistCtx, AssistId, TextRange, TextUnit};
+use hir::{db::HirDatabase, HasSource};
+use ra_db::FilePosition;
+use ra_syntax::{
+    ast::{self, AstNode, NameOwner, TypeAscriptionOwner},
+    T,
+};
+
+pub(crate) fn add_new(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let impl_node = ctx.node_at_offset::<ast::ImplBlock>()?;
+    let item_list = impl_node.item_list()?;
+    let items: Vec<_> = item_list.impl_items().collect();
```

Comment:
I think there's no need to collect to `Vec` here, you can work with iterator directly and avoid allocation that way.